wordpress-mobile/WordPress-Android
# Fix "Edit your homepage" Quick Start task completion

Fixes #14016

This PR basically updates the `QuickStartRepository::requestNextStepOfTask` method so that it doesn't clear the currently active task when called. The reason it used to do that, was because it was originally implemented to be used in conjunction with the `QuickStartUtils::completeTaskAndRemindNextOne` method. However, if we tried to use it in conjunction with `QuickStartRepository::completeTask` it would not work, as that method needs the active task to not be `null`, which was not very intuitive.

One problem with not clearing the currently active task is that if the user doesn't complete the next step of the task and comes back to the My Site fragment, they would still see the current active task focus point, which is not desirable. To fix that, this PR also makes sure to clear the active task inside `QuickStartRepository::refresh` which is always called when the My Site fragment is resumed. This also prevents a focus point from incorrectly appearing after switching sites. Let me know what you think! 

### To test

1. Turn on the Improved My Site flag and restart the app.
1. Go to the My Site screen and select a site with a "Homepage" where Quick Start is active.
1. Under the "Customize your site" category, tap the "Edit your homepage" task.
1. Notice that a focus point appears over the Pages button.
1. Tap on the Pages button.
1. On the Pages screen, notice the focus point over the "Homepage" item.
1. Without clicking anything, go back to the previous screen.
1. On the My Site screen, notice that the focus point has disappeared.
1. Scroll to the "Customize your site" category.
1. Notice that the "Edit your homepage" task is still uncompleted. Tap it again.
1. Repeat steps 4, 5 and 6.
1. On the Pages screen, tap the "Homepage" item.
1. Go back to the My Site screen and scroll to the "Customize your site" category.
1. Notice that the "Edit your homepage" task is now completed.

Also make sure all other tasks are still working as expected.

---

PR submission checklist:
